pip/req/req_file.py

```diff
@@ -51,7 +51,8 @@
 # options to be passed to requirements
 SUPPORTED_OPTIONS_REQ = [
     cmdoptions.install_options,
-    cmdoptions.global_options
+    cmdoptions.global_options,
+    cmdoptions.hash,
 ]
```

Comment:
I think you should add `cmdoptions.require_hashes` to the `SUPPORTED_OPTIONS list above

I assume people might want to put the `--require-hashes` flag in the requirements file at the top like other supported options.